project/src/js/ui/tournament/detail/tournamentDetailView.js

```diff
@@ -43,49 +57,80 @@ function tournamentContentFinished(data) {
   );
 }
 
-function tournamentContentCreated(data) {
+function tournamentContentCreated(ctrl) {
+  let data = ctrl.tournament();
   return (
     <div>
       { tournamentHeader(data, data.secondsToStart, 'Starts in:')}
+      { tournamentJoinWithdraw(ctrl) }
       { tournamentLeaderboard(data, false) }
     </div>
   );
 }
 
-function tournamentContentStarted(data) {
+function tournamentContentStarted(ctrl) {
+  let data = ctrl.tournament();
   return (
     <div>
       { tournamentHeader(data, data.secondsToFinish, '')}
+      { tournamentJoinWithdraw(ctrl) }
       { tournamentLeaderboard(data, false) }
       { data.featured ? tournamentFeaturedGame(data) : '' }
     </div>
   );
 }
 
 function tournamentHeader(data, time, timeText) {
-  let variant = variantInfo(data);
+  let variant = variantDisplay(data);
   let control = timeControl(data);
   return (
-    <div className='basicTournamentInfo'>
-      <strong> {variant + ' • ' + control + ' • ' + data.minutes + 'M' } </strong>
-      <div className='timeInfo'>
-        <strong> {timeInfo(time, timeText)} </strong>
-      </div>
-    </div>
+    <div className='tournamentInfoTime'>
+     <strong className='tournamentInfo' data-icon={gameIcon(variantKey(data))} > {variant + ' • ' + control + ' • ' + data.minutes + 'M' } </strong>
+     <div className='timeInfo'>
+       <strong> {timeInfo(time, timeText)} </strong>
+     </div>
+   </div>
   );
 }
 
-function variantInfo(data) {
-  let variant = data.variant;
-  if (variant === 'standard') {
-    if (data.schedule)
-      variant = data.schedule.speed;
-    else if (data.position)
-      variant = data.position.name;
+function tournamentJoinWithdraw(ctrl) {
+  let label = buttonLabel(ctrl);
+
+  function buttonAction () {
+    if (ctrl.hasJoined())
+      ctrl.withdraw();
     else
-      variant = '';
+      ctrl.join();
   }
 
+  if (ctrl.tournament().variant === 'crazyhouse')
+    return ('');
```

Comment:
Don't return a empty string. return `null` if nothing to display.

Instead of filtering the variant like that you should use the `availableVariants` array in settings.js